File: C++ Programs/Algorithms/SortingAlgo/SelectionSort.cpp
#include <iostream>
#include <vector>
using namespace std;

void disp(vector<int> arr)
{
    cout << "Sorted Array : ";
    for (int i = 0; i < arr.size(); i++)
    {
        if (i < (arr.size() - 1))
            cout << arr[i] << ", ";
        else
            cout << arr[i];
    }
}

void selection_sort(vector<int> arr)
{
    int temp;
    for (int i = 0; i < arr.size()-1; i++)
    {
        int min = i;
        for (int j = i+1; j < arr.size(); j++)
        {
            if (arr[j] < arr[min])
                min = j;
        }
        temp = arr[min];
        arr[min] = arr[i];
        arr[i] = temp;
    }
    disp(arr);
}

int main()
{   
    int size, data;
    vector<int> list;
    cout << "Enter size of the list : ";
    cin >> size;
    cout << "Enter elements for the list (Unordered) : ";
    for (int i = 0; i < size; i++)
    {
        cin >> data;
        list.push_back(data);
    }
    cout << "The sorted list is :" << endl;
    selection_sort(list);
    return 0;
}

File: Java_Programs/Algorithms/SortingAlgo/SelectionSort.java
package Java_Programs.Algorithms.SortingAlgo;

import java.util.Scanner;

public class SelectionSort {

    static void selectionSort(int arr[]) {
        int temp;
        for (int i = 0; i < arr.length; i++) {
            int min = arr[i];
            int minIndex = i;
            for (int j = i; j < arr.length; j++) {
                if (arr[j] < min) {
                    min = arr[j];
                    minIndex = j;
                }
            }
            temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }

        disp(arr);
    }

    static void disp(int arr[]) {
        int n = arr.length;
        for (int i = 0; i < n; ++i)
            System.out.print(arr[i] + " ");
        System.out.println();
    }

    public static void main(String[] args) {
        Scanner x = new Scanner(System.in);
        System.out.print("Enter size of array : ");
        int size = x.nextInt();
        int arr[] = new int[size];
        System.out.print("\nEnter elements : ");
        for (int i = 0; i < size; i++) {
            arr[i] = x.nextInt();
        }

        System.out.println("Sorting...");
        selectionSort(arr);
        x.close();
    }    
}
